x-pack/plugins/observability/public/pages/slo_edit/components/custom_kql/custom_kql_indicator_type_form.tsx

```diff
@@ -18,36 +19,46 @@ import { QueryBuilder } from '../common/query_builder';
 import { IndexSelection } from '../custom_common/index_selection';
 
 export function CustomKqlIndicatorTypeForm() {
-  const { watch } = useFormContext<CreateSLOForm>();
+  const { watch, setValue } = useFormContext<CreateSLOForm>();
   const index = watch('indicator.params.index');
   const { isLoading: isIndexFieldsLoading, data: indexFields = [] } =
     useFetchIndexPatternFields(index);
   const timestampFields = indexFields.filter((field) => field.type === 'date');
   const partitionByFields = indexFields.filter((field) => field.aggregatable);
 
+  const { data: dataViews = [] } = useFetchDataViews({
+    name: index,
+  });
+
+  const dataViewTimeFieldName = dataViews[0]?.timeFieldName;
+
+  useEffect(() => {
+    if (dataViewTimeFieldName) {
+      setValue('indicator.params.timestampField', dataViewTimeFieldName);
+    }
+  }, [dataViewTimeFieldName, setValue, index]);
```

Comment:
I wonder if we create an SLO using an index which has more than one timestamp field, selecting for example the one that is not set on the dataview timestampfield, and we try to edit this SLO. 

Would we prefill the form with the dataview timestampfield instead of the previously selected timestamp field?

Reply:
yeah i think that will happen. i thought about this idea. maybe i can separate the PR to brainstorm further